hudi-common/src/main/java/org/apache/hudi/common/util/ConfigUtils.java

```diff
@@ -79,12 +79,12 @@ public class ConfigUtils {
    */
   public static String getOrderingField(Properties properties) {
     String orderField = null;
-    if (properties.containsKey(HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP_KEY)) {
-      orderField = properties.getProperty(HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP_KEY);
-    } else if (properties.containsKey("hoodie.datasource.write.precombine.field")) {
+    if (properties.containsKey("hoodie.datasource.write.precombine.field")) {
       orderField = properties.getProperty("hoodie.datasource.write.precombine.field");
     } else if (properties.containsKey(HoodieTableConfig.PRECOMBINE_FIELD.key())) {
       orderField = properties.getProperty(HoodieTableConfig.PRECOMBINE_FIELD.key());
+    } else if (properties.containsKey(HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP_KEY)) {
```

Comment:
Is the change in order of these configs intentional? 
previously it was HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP_KEY, followed by "hoodie.datasource.write.precombine.field", and then HoodieTableConfig.PRECOMBINE_FIELD.key().

which ever config property is set, we take that. 

and now, we are switching the ordering of these to 

hoodie.datasource.write.precombine.field -> HoodieTableConfig.PRECOMBINE_FIELD.key() -> HoodiePayloadProps.PAYLOAD_ORDERING_FIELD_PROP_KEY 

ordering field is a immutable table property similar to table type for instance. 
So, why honor a diff write property if set? just in this code snippet. 
Anyways, down the line the write fail if someone attempted to write to a table by switching ordering field config property.

Reply:
> So, why can't we just rely on only the table property here and ignore other options.

we want to keep backward compatibility here, we can remove all the others except the others when the write config got removed.